src/averageAges.js

```diff
@@ -55,7 +75,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  // find women with kids
+  const womenWithChildren = people.filter((child) => {
+    return people.find((woman) => woman.name === child.mother);
+  });
+
+  // find kids only if kids.sex === 'm'
+  const childrenOnlyMens = people.filter((child) => {
+    return (
+      people.find((woman) => woman.name === child.mother) && child.sex === 'm'
+    );
+  });
+
+  const isOnlyWithSon = onlyWithSon ? childrenOnlyMens : womenWithChildren;
```

Comment:
you can simplify this 'block' and avoid creating redundant variables using the ternary operator